plugins/inputs/aos/aos.go

```diff
@@ -335,6 +336,37 @@ func (ssl *streamAos) msgReader(r io.Reader) {
 				}
 			}
 
+
+			if newProbeMessage != nil {
```

Comment:
@huttmf I recall we were going to add test code to this? This is a classic change that would benefit from a unit-test..

Reply:
@Ragsboss @huttmf Added unit tests now. 

I am still working on polishing it further by supporting msg reader .. but it requires structuring of the messages in groups and a few things like that. Will take time.